### PR TITLE
Use reanimatedv2 API in our components

### PR DESCRIPTION
**Pull request by ethanshar**
## Description
Use reanimated v1 & v2 API in order to support both APIs and allow users to gradually migrate to reanimated v2

## Changelog
Begin transition to reanimated v2 in our components, at the moment our components should work on both versions.

### CODE REVIEW

**Review by M-i-k-e-l — COMMENTED** [review]
I liked the backwards compatibility with the `||`!

You didn't need the change in `TabBarItem` (`processColor` etc)? I think I had a bug without it.

File: src/incubator/WheelPicker/index.tsx [hunk 1]
```diff
@@ -107,6 +107,7 @@ const WheelPicker = ({items, itemHeight = 48, activeTextColor, inactiveTextColor
       <View width={250} height={height} br20>
         <AnimatedFlatList
           data={items}
+          // @ts-expect-error
```
[the remark this answers is not in the document]
Comment (ethanshar) [reply src/incubator/WheelPicker/index.tsx:110]: ohh bummer, ok ill change

[the remark this answers is not in the document]
Comment (ethanshar) [reply src/incubator/WheelPicker/index.tsx]: fixed

**Comment by ethanshar** [comment]
> You didn't need the change in `TabBarItem` (`processColor` etc)? I think I had a bug without it.

I didn't notice something weird with it. I think it worked fine

**Comment by M-i-k-e-l** [comment]
> I didn't notice something weird with it. I think it worked fine

Maybe they fixed it, approved, not merged

**Review by M-i-k-e-l — APPROVED** [review]

**Pull request merged**